Issue #3222538 by vnech: Allow "Content Manager" role to translate content

nechai opened the pull request:
## Problem
In OS "Content manager" role doesn't have permissions to add/edit/delete translations but has permissions to manage most of the content.

## Solution
Grand "Content manager" role with permissions to translate content he has permissions to edit.

## Issue tracker
- https://www.drupal.org/project/social/issues/3222538
- https://getopensocial.atlassian.net/browse/YANG-5978
- https://getopensocial.atlassian.net/browse/YANG-5985

## How to test
- [ ] Make sure your site is multilingual
- [ ] Login as CM
- [ ] Create an Event
- [ ] Try to add a translation to the created event

## Release notes
*Allow "Content Manager" role to translate content.*

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

ribel commented:
Hey, @ronaldtebrake and @Kingdutch!
It seems logical to allow CM to translate the content, but earlier it was set only to SM.
Want to confirm with you if this is ok to make this change, and if it is ok to have it in 10.2.2?

Kingdutch commented:
>  It seems logical to allow CM to translate the content, but earlier it was set only to SM.

This is more a product related question that I think Moritz or a product designer should answer. It depends on what our plans are with content translation.

>  Want to confirm with you if this is ok to make this change, and if it is ok to have it in 10.2.2?

In my opinion it's a change in behaviour and not a fix for something that was broken. That would classify it as a feature rather than a bugfix. So that would mean it should go in 10.3.0. However, I'll defer to Ronald's judgement on that one.

Kingdutch commented on modules/custom/social_language/social_language.install:
```diff
@@ -114,3 +117,11 @@ YAML;
+ */
+function social_language_update_10301() {
+  // The function grand permissions.
+  _social_language_set_permissions();
```
Update hooks should [be explicit](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Extension%21module.api.php/function/hook_update_N/8.2.x) in changing from one state to another. The `_social_language_set_permissions` function is used to set the module's default permissions and may change over time. This makes the update hook unstable.

nechai replied:
Added changes

ronaldtebrake commented:
> > It seems logical to allow CM to translate the content, but earlier it was set only to SM.
> 
> This is more a product related question that I think Moritz or a product designer should answer. It depends on what our plans are with content translation.
> 
> > Want to confirm with you if this is ok to make this change, and if it is ok to have it in 10.2.2?
> 
> In my opinion it's a change in behaviour and not a fix for something that was broken. That would classify it as a feature rather than a bugfix. So that would mean it should go in 10.3.0. However, I'll defer to Ronald's judgement on that one.



> > It seems logical to allow CM to translate the content, but earlier it was set only to SM.
> 
> This is more a product related question that I think Moritz or a product designer should answer. It depends on what our plans are with content translation.
> 
> > Want to confirm with you if this is ok to make this change, and if it is ok to have it in 10.2.2?
> 
> In my opinion it's a change in behaviour and not a fix for something that was broken. That would classify it as a feature rather than a bugfix. So that would mean it should go in 10.3.0. However, I'll defer to Ronald's judgement on that one.

Couldn't agree more. 10.3.0 would be best indeed for this unless we can prove it's definitely a bug. (I couldnt find anything yet even if it makes sense to me a CM could add translations) but we don't want others to be forced in this in a patch release if we can't provide input as to why it's a bug and not a feature.

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

nechai commented:
@ribel In the beginning I added changes to grand CM to translate any content. But I found that the CM does not always have permissions to edit any content (for example, CM can't edit "Flexible Group" if he is not an author or group manager). 
So, I added changes that allow CM to add/edit/delete translations that he can edit. Probably, we need to change PR title 🤔

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63

open-social-tugboat commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2404-2gsymbrkuu5oljk0kdiiwnndoytuyvie.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60e58d75f543e75ce66c4f63
